test/unit/conditions/context.test.ts

```diff
@@ -90,20 +90,16 @@ describe('context parameters', () => {
     };
     const condition = new ContractCondition(conditionObj);
     const conditionExpr = new ConditionExpression(condition);
-    const conditionContext = conditionExpr.buildContext(provider, {}, signer);
     expect(conditionExpr.contextRequiresSigner()).toBe(true);
-    expect(conditionContext.requiresSigner()).toBe(true);
   });
 
   it('detects if a signer is not required', () => {
     const condition = new RpcCondition(testRpcConditionObj);
     const conditionExpr = new ConditionExpression(condition);
-    const conditionContext = conditionExpr.buildContext(provider, {}, signer);
```

Comment:
Similar comment as above for completeness. After the `contextRequiresSigner` check on L105, you can ensure that context can be successfully built with/without a signer since one isn't needed.